docs/admin/migrations.rst

```diff
@@ -0,0 +1,40 @@
+Migrations
+==========
+
+From Jenkins and GitLab to Jenkins and LocalVC
+----------------------------------------------
+
+Here you can find the steps to migrate from a :ref:`GitLab and Jenkins <Jenkins and GitLab Setup>`
+to a Jenkins and LocalVC Setup.
+
+0. Your Artemis is configured to run with Jenkins and GitLab and is currently not running.
+   Artemis should not become reachable for users when starting it for the migration (e.g., by shutting down the nginx webproxy in front of it) to make sure no other changes are made during the migration.
+1. Make a Backup of your Jenkins and Artemis data.
+2. Adjust parameters in a loaded `application-*.yml`:
+
+   1. Set in ``artemis.version-control.local-vcs-repo-path`` the path where LocalVC should store the repositories.
+   2. Adjust the scaling parameters under ``migration.scaling`` if you prefer other values then the defaults.
+      The migrations consist of two sequential steps (GitLab and Jenkins).
+      In each of them several threads process the repositories. Each batch is processed by one thread.
+
+      1. ``batch-size`` (default 10): Number of repositories that are loaded in one batch.
+      2. ``max-thread-count`` (default 4): Maximal number of concurrently running threads for the migration.
+      3. ``timeout-in-hours`` (default 48): The timeout value for each step of the migration.
+      4. ``estimated-time-per-repository`` (default 2): Time in seconds,
+         which is used to give an overall estimation for one step and to log an estimated percentage progress.
+3. Start Artemis with the additional profile ``migrate-gitlab-jenkins-to-localvc`` to perform the migration.
+   As a minimum, the profiles ``core,gitlab,jenkins,migrate-gitlab-jenkins-to-localvc`` are required.
+   Additional profiles may be required for your setup (e.g., profile ``prod`` if you have additional configuration in ``application-prod.yml``).
+   You should not yet add ``localvc``.
+   After it Artemis stops automatically (with a ``Application run failed`` message).
+   Check the log of Artemis for entries with ``Migration`` to see if there were any problems.
+4. Adjust the credential on your Jenkins, which was previously used to :ref:`access your GitLab Repositories <Jenkins GitLab Repository Access>`.
+   The same credential ID now have to provide username and password of an administrator account of your Artemis instance
+   or dedicated git credential configured in ``build-agent-git-username`` and ``build-agent-git-password`` under ``artemis.version-control``.
+5. Change your Artemis configuration under ``artemis.version-control`` to run with LocalVC instead of GitLab.
+
+   1. Adjust ``url`` to point to your Artemis instance instead of your GitLab instance.
+      (Same adjustment for ``ssh-template-clone-url`` if present.)
+   2. Remove all entries for the authentication (i.e. ``user``, ``password`` and ``token``).
```

Comment:
Are you sure?

I'm running artemis with just the following on our test system

```
version-control:
    build-agent-git-password: password
    build-agent-git-username: jenkins
    ssh-port: 22
    url: https://url
```

Reply:
Hm, it failed with a missing dependency for that value. I’ll check.

I don’t think it will cause problems if the username remains in the config after the migration in any case. So I would say this is not a blocker and the PR can be merged with or without mentioning `user` in the docs here.

Reply:
Agree :) I was just wondering